### ISSUE TITLE
bio_dgram_test fails if IPV6 disabled/absent

### ISSUE THREAD

**Issue by Tarnum-tst** [openssl/openssl#19142]
Using latest commit from master branch. Building on Windows / MSYS2 / MinGW. If IPV6 disabled, got this:

```
04-test_bio_dgram.t  ................
    #  INFO:  @ test/bio_dgram_test.c:127
    # # Testing with AF_INET6, local=0
    #
    # ERROR: (int) 'fdl >= 01' failed @ test/bio_dgram_test.c:167
    # [-1] compared to [0]
    # C4100000:error:8000273F:system library:BIO_socket:reason(10047):crypto/bio/bio_sock2.c:51:calling socket()
    # C4100000:error:10000076:BIO routines:BIO_socket:unable to create socket:crypto/bio/bio_sock2.с:53:
    # OPENSSL_TEST_RAND_ORDER=1662120827
    not ok 2 - iteration 2
# ---------------------------------------------------------------------------------------------------------------
    # INFO:  @ test/bio_dgram_test.c:127
    # # Testing with AF_INET6, local=1
    #
    # ERROR: (int) 'fdl >= 0' failed @ test/bio_dgram_test.c:167
    # [-1] compared to [0]
    # C4100000:error:8000273F:system library:BIO_socket:reason(10047):crypto/bio/bio_sock2.c:51:calling socket()
    # C4100000:error:10000076:BIO routines:BIO_socket:unable to create socket:crypto/bio/bio_sock2.с:53:
    # OPENSSL_TEST_RAND_ORDER=1662120827
    not ok 4  - iteration 4
# ---------------------------------------------------------------------------------------------------------------
    # OPENSSL_TEST_RAND_ORDER=1662120827
    not ok 1 - test_bio_dgram
#----------------------------------------------------------------------------------------------------------------
../../util/wrap.pl ../../test/bio_dgram_test.exe => 1
not ok 1 -  running bio_dgram_test
04-test_bio_dgram.t  ................ 1/? -----------------------------------------------------------------------
#   Failed test  'running bio_dgram_test'
#   at /home/User/openssl-master/util/perl/OpenSSL/Test/Simple.pm line 77.
04-test_bio_dgram.t ................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests
```
I believe, checking in bio_dgram_test.c for the presence of (enabled) IPV6 in the user system would not hurt. Thanks!

**Comment by t8m** [comment]
@hlandau Can you please look?

**Comment by hlandau** [comment]
Should be fixed by #19181
